### DISCUSSION TITLE
Recuring Job only called once in Blazor Hybrid App

### DISCUSSION THREAD

**Discussion by konrad79** [shinyorg/shiny#1488] [Q&A] [answered]
I've a Blazor Hybrid App with Shiny and .NET8.
I want a background task run recurrently every N seconds or minutes.

I registered the job in the Programm.cs:

  ```
public static MauiApp CreateMauiApp()
{
   var builder = MauiApp.CreateBuilder();
   builder
      .UseMauiApp<App>()
      .UseShiny()
      .ConfigureFonts(fonts =>
      {
         fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
      });

   builder.Services.AddMauiBlazorWebView();
            
#if ANDROID
   var job = new JobInfo("MyJobName", typeof(MyJob), RunOnForeground: false, null , RequiredInternetAccess: InternetAccess.Any, DeviceCharging: false, BatteryNotLow: true, IsSystemJob: false);            
   builder.Services.AddJob(job);
#endif

#if DEBUG
   builder.Services.AddBlazorWebViewDeveloperTools();
   builder.Logging.AddDebug();
#endif
   return builder.Build();
}
```

(I've to use the #if ANDROID because of errors otherwise)

And in the Job itself I defined the MinimumTime (no documentation found for this parameter - hoping it's the right one):

```
public class MyJob : Job
{
    public MyJob(ILogger<MyJob> logger) : base(logger)
    {            
        this.MinimumTime = TimeSpan.FromSeconds(10);
    }

    protected override async Task Run(CancellationToken cancelToken)
    {
        // DO SOMETHING LIKE CALLING AN API AND SHOW NOTIFICATION
    }
}
```

The code in the Run function is only called once at the startup of the app.

What am I missing?

And are there older versions, because I found a lot of examples using Job parameters like "Repeat = true", "PeriodicTime = ..." and others, that doesn't exist in Shiny.NET for .NET8 (anymore?).

**Comment by aritchie** [comment, marked as answer]
Jobs don't work like that.  Periodic != Scheduled.  The OS truly decides how often things run.  MinimumTime is used to ensure it doesn't run too often.

https://shinylib.net/client/jobs/faq/


> And are there older versions, because I found a lot of examples using Job parameters like "Repeat = true", "PeriodicTime = ..." and others, that doesn't exist in Shiny.NET for .NET8 (anymore?).

The current samples don't have this.  Docs are behind though.

**Reply by konrad79** [reply]
Thanks, Allan.
Then I can say goodbye to my idea of having a background process call an API every 5 minutes - and then having a push notification pop up under certain conditions
